tcms/testcases/views.py

```diff
@@ -445,21 +444,23 @@ def list_all(request):  # pylint: disable=missing-permission-required
     return render(request, 'plan/get_cases.html', context_data)
 
 
-@require_GET
-def search(request):  # pylint: disable=missing-permission-required
+class TestCaseSearchView(TemplateView):  # pylint: disable=missing-permission-required
     """
-        Shows the search form which uses JSON RPC to fetch the resuts
+        Shows the search form which uses JSON RPC to fetch the results
     """
-    form = SearchCaseForm(request.GET)
-    if request.GET.get('product'):
-        form.populate(product_id=request.GET['product'])
-    else:
-        form.populate()
 
-    context_data = {
-        'form': form,
-    }
-    return render(request, 'testcases/search.html', context_data)
+    template_name = 'testcases/search.html'
+
+    def get_context_data(self, **kwargs):
+        form = SearchCaseForm(self.request.GET)
+        if self.request.GET.get('product'):
+            form.populate(product_id=self.request.GET['product'])
+        else:
+            form.populate()
+
+        return {
+            'form': form,
+        }
```

Comment:
FYI when syntax allows you can also stuff that expression on the same line. It doesn't hurt readability.